homeassistant/components/rpi_gpio.py

```diff
@@ -48,6 +48,16 @@ def setup_input(port, pull_mode):
                GPIO.PUD_DOWN if pull_mode == 'DOWN' else GPIO.PUD_UP)
 
 
+def run_pwm(port, frequency, duty_cycle, duration):
+    """Run GPIO as PWM."""
+    import RPi.GPIO as GPIO
+    import time
+    p = GPIO.PWM(port, frequency)
+    p.start(duty_cycle)
+    time.sleep(duration)
```

Comment:
Don't use `time.sleep`. Use `track_point_in_time` from `helpers.event`. You need access to `hass` for that so you could add `hass` as a parameter in the signature of `run_pwm`. Eg:
```py
after_duration = datetime.now() + duration
def stop(*args):
    """Call stop."""
    p.stop()
track_point_in_time(hass, stop, after_duration)
```

Reply:
Okay, I guess i need to change  _run_with_duty(self, duty): to reflect these changes too, since it needs to wait for the stop command before running (i.e. it does two calls in sequence right now and can't handle a async operation).